File: 4779/4779.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   4779.cpp                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2025/07/12 15:51:11 by jihoolee          #+#    #+#             */
/*   Updated: 2025/07/12 16:45:57 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <string>
#include <cmath>

std::string cantor(int num) {
  if (num == 0) {
    return "-";
  } else {
    std::string side = cantor(num - 1);
    std::string center = "";

    for (size_t idx = 0; idx < pow(3, num - 1); ++idx)
      center += " ";
    return side + center + side;
  }
}

int main(void) {
  int num;

  while (std::cin >> num) {
    std::cout << cantor(num) << std::endl;
  }
  return 0;
}

File: 4779/4779.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    4779.py                                            :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2025/07/12 15:18:30 by jihoolee          #+#    #+#              #
#    Updated: 2025/07/12 15:50:15 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def cantor(num: int) -> str:
    if num == 0:
        return '-'
    else:
        side: str = cantor(num - 1)
        center: str = ' ' * (3 ** (num - 1))
        return side + center + side


def main() -> None:
    while True:
        try:
            num: int = int(input())
            print(cantor(num))
        except EOFError:
            break


if __name__ == "__main__":
    main()
